src/lib.rs

```diff
@@ -50,12 +50,22 @@ impl<Data: AsRef<[u8]>> Reader<Data> {
         if !input.as_ref().starts_with(&KTX2_MAGIC) {
             return Err(ParseError::BadMagic);
         }
-        let header = input.as_ref()[0..Header::LENGTH].try_into().unwrap();
-        Header::from_bytes(header).validate()?;
+        let header_data = input.as_ref()[0..Header::LENGTH].try_into().unwrap();
+        let header = Header::from_bytes(header_data);
+        header.validate()?;
+
+        // Check data format descriptor integrity
+        let mut offset = header.dfd_byte_offset as usize;
+        let length = bytes_to_u32(input.as_ref(), &mut offset)?;
+        if length != header.dfd_byte_length
```

Comment:
`UnexpectedEnd` is a confusing error to return for what is actually an inconsistent/illegal field value. We could add a new error case, but given that the spec guarantees that these values are equal, I'd also be comfortable just omitting the check; worst-case scenario is that a malformed file yields garbage data, which is always possible regardless.

Reply:
In that case I guess I can just sanity check the fields used to slice the data to make sure they're in-bounds. As I use `dfd_byte_offset + dfd_byte_length` as then end, I could just check that the length of the data is >= that. Then returning `UnexpectedEnd` makes sense to me.

Reply:
Yeah, sounds sensible.